L120-requiring-cpp17.md

```diff
@@ -0,0 +1,29 @@
+L120: Requiring C++17 in gRPC Core/C++ Library
+----
+* Author(s): veblush
+* Approver: markdroth
+* Status: Draft
+* Implemented in: n/a
+* Last updated: Oct 14, 2024
+* Discussion at: TBD
+
+## Abstract
+
+gRPC starts requiring C++17.
+
+## Background
+
+To leverage the advancements in C++ standards, gRPC has progressively updated its requirements.  
+Initially, it adopted C++11 in 2017 (as per [Allow C++ in gRPC Core Library](L6-core-allow-cpp.md)).
```

Comment:
Please add "L6:" to the start of the text for this link, to make it clear that it references a gRFC.

Reply:
Done.